src/sentry/migrations/0572_rm_cficache_symcache.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 3.2.20 on 2023-10-05 09:51
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+# There is a bunch of orphaned `File`s still being stored that will be cleaned up.
+# They have the following `type`s:
+# - `project.symcache` and `project.cficache` have not been created ever since
+#   <https://github.com/getsentry/sentry/commit/c14c8f0e3a30fb5c2221fdefa4c0f5167bdc8986#diff-08e824e030b14e7936ffeda41ccf81d8feed2cdcdfbc522d0c46d0064f3640c9L750>
+# - `global.dsym` seems to be leftover from 2016
+# - `replay.recording` were moved to a different backing store.
+TYPES_TO_DELETE = ("project.symcache", "project.cficache", "global.dsym", "replay.recording")
+
+
+def rm_cficache_symcache(apps, schema_editor):
+    File = apps.get_model("sentry", "File")
+
+    # According to <https://develop.sentry.dev/database-migrations/#filters>:
+    # > it is better to iterate over the entire table instead of using a filter
+    # Also, there is no index on `type` anyway.
+    for file in RangeQuerySetWrapperWithProgressBar(File.objects.filter(type__in=TYPES_TO_DELETE)):
+        if file.type in TYPES_TO_DELETE:
```

Comment:
To clarify, are we doing `File.objects.filter`, or the full scan (what we had before, and what's explained in the comment above)? Also, `if file.type in TYPES_TO_DELETE` seems to be redundant now.

The problem I can think of, under the hood `RangeQuerySetWrapperWithProgressBar` calls `queryset.count()`, which will basically do the full scan anyway (with filtering or not).
I can see now that there's also `RangeQuerySetWrapperWithProgressBarApprox` that doesn't get an exact count. We can probably use that along with the `File.objects.all()`.